pkg/api/staking_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	stakingContractMock "github.com/ethersphere/bee/pkg/staking/stakingcontract/mock"
+)
+
+func TestDepositStake(t *testing.T) {
+	k, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		t.Fatal(err)
+	}
+	addr, err := crypto.NewOverlayAddress(k.PublicKey, 1, common.HexToHash("0x1").Bytes())
+	if err != nil {
+		t.Fatal(err)
+	}
+	minStake := big.NewInt(1).String()
+	minStakedAmount := big.NewInt(1)
```

Comment:
`minStakedAmount` value should come from constant `stakingcontract.MinimumStakeAmount`